src/pydvl/valuation/samplers/utils.py

```diff
@@ -10,7 +10,7 @@
 T = TypeVar("T")
 
 
-def take_n(it: Iterable[T], size: int = 2) -> Generator[Iterator[T], Any, None]:
+def take_n(it: Iterable[T], size: int = 2) -> Generator[List[T], Any, None]:
```

Comment:
There is a nice little package called more-itertools, see: https://more-itertools.readthedocs.io/en/stable/_modules/more_itertools/more.html#chunked

Just mentioning, in case you did not know it before. The name "take_n" is a bit missleading, as I would expect to get the first "n" (or "size") elements.  Actually, the definition of batch in line 23 is called "take" in more-itertools: https://more-itertools.readthedocs.io/en/stable/_modules/more_itertools/recipes.html#take

Reply:
Yes, `more_itertools.chunked` does exactly what `take_n` did. I replaced it everywhere and removed `take_n`.